### PR TITLE
Give a more user friendly message when plugin program file does not e…

### PR DESCRIPTION
**Pull request by celeron533**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [x] Require translation update
- [ ] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New feature

---

### Description of your *pull request* and other information

At least gives a hint to user to check whether the plugin program file exists on file system.

### CODE REVIEW

File: shadowsocks-csharp/Data/zh_CN.txt [hunk 1]
```diff
@@ -183,6 +183,7 @@ Proxy handshake failed=代理握手失败
 Register hotkey failed=注册快捷键失败
 Cannot parse hotkey: {0}=解析快捷键失败: {0}
 Timeout is invalid, it should not exceed {0}=超时无效，不应超过 {0}
+Cannot find the plugin program file=找不到插件程序文件
```
Comment (studentmain) [inline shadowsocks-csharp/Data/zh_CN.txt]: Add program name here?

Comment (celeron533) [reply shadowsocks-csharp/Data/zh_CN.txt]: File name will be mentioned in the exception it self in `Sip003Plugin.cs`
```
throw new FileNotFoundException(I18N.GetString("Cannot find the plugin program file"), _pluginProcess.StartInfo.FileName, ex);
```

image

File: shadowsocks-csharp/Controller/Service/Sip003Plugin.cs [hunk 2]
```diff
@@ -94,7 +94,18 @@ public bool StartIfNeeded()
...
                 _pluginProcess.StartInfo.Arguments = ExpandEnvironmentVariables(_pluginProcess.StartInfo.Arguments, _pluginProcess.StartInfo.EnvironmentVariables);
-                _pluginProcess.Start();
+                try
+                {
+                    _pluginProcess.Start();
+                }
+                catch (System.ComponentModel.Win32Exception ex)
+                {
+                    // do not use File.Exists(...), it can not handle the scenarios when the plugin file is in system environment path.
+                    if ((uint)ex.ErrorCode == 0x80004005)   // file not found
```
Comment (studentmain) [inline shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: FYI: #2729

Comment (celeron533) [reply shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: Thanks, will use native error code to find out the real problem
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d

Comment (celeron533) [reply shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: But it is uncommon to see "The specified executable is not a valid application for this OS platform." error. Only when run 64 bit program on 32 bit platform. or just have the bad image.

Comment (studentmain) [reply shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: Yes, it's uncommon, but there is😥. Just check if it is FileNotFound, if not, throw error with (or without) native error code so we won't misleading user.

Comment (celeron533) [reply shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: Personally prefer to use current implementation to keep everything simple.

Comment (studentmain) [reply shadowsocks-csharp/Controller/Service/Sip003Plugin.cs]: Copy paste from here
https://docs.microsoft.com/zh-cn/windows/win32/seccrypto/common-hresult-values?redirectedfrom=MSDN


Name | Description | Value
-- | -- | --
S_OK | Operation successful | 0x00000000
E_ABORT | Operation aborted | 0x80004004
E_ACCESSDENIED | General access denied error | 0x80070005
E_FAIL | Unspecified failure | 0x80004005
......|......|.......

Seems we should use native error code...

**Pull request merged**